subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi_gpiote.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Extracted from radio_nrf_dppi.h functions that reference gpiote variables. */
+
+#if defined(HAL_RADIO_GPIO_HAVE_PA_PIN) || defined(HAL_RADIO_GPIO_HAVE_LNA_PIN)
+static inline void hal_palna_ppi_setup(void)
+{
+	nrf_timer_publish_set(EVENT_TIMER, NRF_TIMER_EVENT_COMPARE2,
+			      HAL_ENABLE_PALNA_PPI);
+	nrf_radio_publish_set(NRF_RADIO, NRF_RADIO_EVENT_DISABLED,
+			      HAL_DISABLE_PALNA_PPI);
+
+#if !defined(HAL_RADIO_FEM_IS_NRF21540)
+	nrf_gpiote_task_t task;
+
+	task = nrf_gpiote_out_task_get(gpiote_ch_palna);
+	nrf_gpiote_subscribe_set(gpiote_palna.p_reg, task, HAL_DISABLE_PALNA_PPI);
+#endif
+}
+#endif /* defined(HAL_RADIO_GPIO_HAVE_PA_PIN) || defined(HAL_RADIO_GPIO_HAVE_LNA_PIN) */
+
+/******************************************************************************/
+#if defined(HAL_RADIO_FEM_IS_NRF21540)
+
+static inline void hal_pa_ppi_setup(void)
+{
+	nrf_gpiote_task_t task;
+
+#if defined(HAL_RADIO_GPIO_PA_POL_INV)
+	task = nrf_gpiote_clr_task_get(gpiote_ch_palna);
```

Comment:
it would be worth to create a helper function and avoid code duplication:
```
static inline void gpiote_ppi_setup(NRF_GPIOTE_Type*reg, int pin, uint8_t ppi, bool inv);
```

Reply:
Done.